### PR TITLE
feat: wrap gas limit into a new type

### LINKED ISSUE #3813: Put rpc gas cap in EthApi in a new type
### Describe the feature

the configurable rpc gas cap accepts an u64

https://github.com/paradigmxyz/reth/blob/eb32fd3c6d2feec5e845981ecbe91139656e6bff/crates/rpc/rpc/src/eth/api/mod.rs#L79-L86

which should default to 50M: see https://github.com/paradigmxyz/reth/pull/3812

to make this easier we should put this in a new type so you can invoke it via `Default::default()`

## TODO

add new type wrapper for gas cap in rpc module
change `new` to accept `impl Into<GasCap>`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by altugbakan**
Fixes #3813

I tried to create the new type, but I'm not sure if I got everything right. Can you please tell me if I'm missing something?

Thanks!

### CODE REVIEW

File: crates/rpc/rpc/src/eth/api/mod.rs [hunk 1]
```diff
@@ -370,6 +370,22 @@ where
     }
 }
 
+/// The wrapper type for gas limit
+#[derive(Debug, Clone, Copy)]
+pub struct GasCap(u64);
```
Comment (Rjected) [inline crates/rpc/rpc/src/eth/api/mod.rs:375]: this should also have a `Default` impl for the 50M default

**Review by Rjected — COMMENTED** [review]
good start! just a reminder about the default so far

File: crates/rpc/rpc/src/eth/api/mod.rs (hunk 1 above)
Comment (altugbakan) [reply crates/rpc/rpc/src/eth/api/mod.rs:375]: Should I use the ETHEREUM_BLOCK_GAS_LIMIT or just hardcode the value instead?

Comment (Rjected) [reply crates/rpc/rpc/src/eth/api/mod.rs:375]: You could use this:
https://github.com/paradigmxyz/reth/blob/5ad9b32cbcbb9ed1e08a645674731cc78a0f7ed3/crates/rpc/rpc-builder/src/constants.rs#L10-L15

Comment (altugbakan) [reply crates/rpc/rpc/src/eth/api/mod.rs:375]: I tried using that but I couldn't import the `rpc-builder` crate due to cyclic dependencies occurring. Can you give me some tips about how to solve this?

Comment (Rjected) [reply crates/rpc/rpc/src/eth/api/mod.rs:375]: In that case I would move the `RPC_DEFAULT_GAS_CAP` from `rpc/rpc-builder/src/constants.rs` to `rpc/rpc/eth/api/mod.rs` maybe? And then it should b exposed in `reth-rpc`, and imported wherever it's used in `rpc-builder`

Comment (altugbakan) [reply crates/rpc/rpc/src/eth/api/mod.rs:382]: Done!

**Comment by altugbakan** [comment]
I merged main into my branch and ran the formatter so the pipeline should pass now.

**Review by mattsse — APPROVED** [review]
lgtm

File: crates/rpc/rpc/src/eth/api/mod.rs [hunk 2]
```diff
@@ -370,6 +370,35 @@ where
     }
 }
 
+/// The default gas limit for eth_call and adjacent calls.
+///
+/// This is different from the default to regular 30M block gas limit
+/// [ETHEREUM_BLOCK_GAS_LIMIT](reth_primitives::constants::ETHEREUM_BLOCK_GAS_LIMIT) to allow for
+/// more complex calls.
+pub const RPC_DEFAULT_GAS_CAP: u64 = 50_000_000;
```
Comment (onbjerg) [inline crates/rpc/rpc/src/eth/api/mod.rs:378]: nit: can this not be `GasCap`?

Comment (altugbakan) [reply crates/rpc/rpc/src/eth/api/mod.rs]: Updated!

**Comment by altugbakan** [comment]
Do I need to fix anything on my side?

**Pull request merged**